pkg/api/deprovision.go

```diff
@@ -142,21 +143,46 @@ func (s *server) deprovision(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	task := model.NewTask(
-		"deprovisionStep",
-		map[string]string{
-			"stepName":   firstStepName,
-			"instanceID": instanceID,
-		},
-	)
+	childCount, err := s.store.GetInstanceChildCountByAlias(instance.Alias)
+	if err != nil {
+		logFields["step"] = firstStepName
+		logFields["error"] = err
+		log.WithFields(logFields).Error(
+			"deprovisioning error: error determining child count",
+		)
+		s.writeResponse(w, http.StatusInternalServerError, responseEmptyJSON)
+	}
+
+	var task model.Task
+	if childCount > 0 {
+		task = model.NewDelayedTask(
+			"waitForChildrenStep",
+			map[string]string{
+				"deprovisionFirstStep": firstStepName,
```

Comment:
For consistency with the more immediate provision/deprovision tasks, can this just be "stepName"?

Reply:
Actually-- strike that. We don't need to pass this information through the queue.

When the step that checks children's statuses wakes up, it retrieves the instance by ID. From that instance, we know the service and we can get the service's provisioner, which can tell us what its first step is.

Reply:
Yeah, this is totally doable. I passed this through just to avoid duplicating the lookup logic from the things in pkg/api (get service, get plan, get service manger, get deprovisioner/provisioner), but this is easily changed.

Reply:
> I passed this through just to avoid duplicating the lookup logic from the things in pkg/api (get service, get plan, get service manger, get deprovisioner/provisioner), but this is easily changed.

I have a todo to look up service and plan when instances are retrieved from the database-- then we can _stop_ doing that in a bunch of other places.

Reply:
👍